pkg/build/api/v1/types.go

```diff
@@ -163,6 +163,9 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
 
+	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
+	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
```

Comment:
Unfortunately there's a very thin line between our BuildSource and *BuildStrateg-ies and moving towards cleaner separation of the two might be beneficial, imho. I'd like to avoid a situation where I need to look at validation to find out which fields should or should not be used jointly. This is not dev nor user friendly.